engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreActor.scala

```diff
@@ -1,242 +1,33 @@
-    } yield source.copyOptions(processOptions(processedWorkflowOptions))
-  }
-
-  private def addWorkCompletionHooks[A](command: WorkflowStoreActorCommand, work: Future[A]) = {
-    work.onComplete {
-      case Success(_) =>
-        self ! WorkDone
-      case Failure(t) =>
-        log.error("Error occurred during {}: {} because {}", command.getClass.getSimpleName, t.toString, ExceptionUtils.getStackTrace(t))
-        self ! WorkDone
-    }
-  }
-
-  /**
-    * Fetches at most n workflows, and builds the correct response message based on if there were any workflows or not
-    */
-  private def newWorkflowMessage(maxWorkflows: Int): Future[WorkflowStoreActorResponse] = {
-    def fetchRunnableWorkflowsIfNeeded(maxWorkflowsInner: Int, state: StartableState) = {
-      if (maxWorkflows > 0) {
-        store.fetchRunnableWorkflows(maxWorkflowsInner, state)
-      } else {
-        Future.successful(List.empty[WorkflowToStart])
-      }
-    }
-
-    val runnableWorkflows = for {
-      restartableWorkflows <- fetchRunnableWorkflowsIfNeeded(maxWorkflows, WorkflowStoreState.Restartable)
-      submittedWorkflows <- fetchRunnableWorkflowsIfNeeded(maxWorkflows - restartableWorkflows.size, WorkflowStoreState.Submitted)
-    } yield restartableWorkflows ++ submittedWorkflows
-
-    runnableWorkflows map {
-      case x :: xs => NewWorkflowsToStart(NonEmptyList.of(x, xs: _*))
-      case _ => NoNewWorkflowsToStart
-    }
-  }
-
-  /**
-    * Takes the workflow id and sends it over to the metadata service w/ default empty values for inputs/outputs
-    */
-  private def registerSubmissionWithMetadataService(id: WorkflowId, originalSourceFiles: WorkflowSourceFilesCollection): Unit = {
-    val sourceFiles = processSource(_.clearEncryptedValues)(originalSourceFiles).get
-
-    val submissionEvents = List(
-      MetadataEvent(MetadataKey(id, None, WorkflowMetadataKeys.SubmissionTime), MetadataValue(OffsetDateTime.now.toString)),
-      MetadataEvent.empty(MetadataKey(id, None, WorkflowMetadataKeys.Inputs)),
-      MetadataEvent.empty(MetadataKey(id, None, WorkflowMetadataKeys.Outputs)),
-
-      MetadataEvent(MetadataKey(id, None, WorkflowMetadataKeys.SubmissionSection, WorkflowMetadataKeys.SubmissionSection_Workflow), MetadataValue(sourceFiles.wdlSource)),
-      MetadataEvent(MetadataKey(id, None, WorkflowMetadataKeys.SubmissionSection, WorkflowMetadataKeys.SubmissionSection_Inputs), MetadataValue(sourceFiles.inputsJson)),
-      MetadataEvent(MetadataKey(id, None, WorkflowMetadataKeys.SubmissionSection, WorkflowMetadataKeys.SubmissionSection_Options), MetadataValue(sourceFiles.workflowOptionsJson))
-    )
-
-    serviceRegistryActor ! PutMetadataAction(submissionEvents)
+  override def receive = {
+    case cmd: WorkflowStoreActorSubmitCommand => workflowStoreSubmitActor forward cmd
+    case cmd: WorkflowStoreActorEngineCommand => workflowStoreEngineActor forward cmd
```

Comment:
Thinking about it more I think read & write *is* the right thing to do here now and the need to separate submit/engine is an artifact of something which no longer the case.

That does raise the question of the dispatcher but at the moment the way our dispatchers are configured i don't think it would matter anyways. I could also define a separate workflow store dispatcher